src/elm/type.coffee

```diff
@@ -112,10 +112,13 @@ module.exports.ToRatio = class ToRatio extends Expression
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined'
+    if arg?
       # Argument will be of form '<quantity>:<quantity>'
       try
-        [numerator, denominator] = arg.toString().split(':').map((quantity) -> parseQuantity(quantity))
+        # String will be split into the following format ["", "numerator", "denominator", ""]
+        splitRatioString = arg.toString().split(/^(\d+\.\d+?\s*\'\w+\'\s*):(\s*\d+\.\d+?\s*\'\w+\'?)/)
+        numerator = parseQuantity(splitRatioString[1])
+        denominator = parseQuantity(splitRatioString[2])
```

Comment:
If the string isn't a valid representation of a ratio, `splitRatioString` (which I suggest should now just be called `match` if you use the `match` function) will be `null`.  If you continue using the `split` function then it won't be null, but just an array of length 1.  In either case, when you try to index into `splitRatioString[1]` you'll either get a dereference error or an out of bounds error.  It will be caught below though.  Is that the intention?  It might be cleaner to do a null-check first instead (and keep the `catch` for if the individual quantities are poorly formed).

But... this one could go either way, so it's up to you.